rfcs/RFC: Handling XLA Exceptions in Python

```diff
@@ -0,0 +1,101 @@
+RFC: Handling XLA Exceptions in Python
+
+Status	Propose
+Author	Vinay Kumar Burugu
+Collaborator	Loki Ravi
+PR	https://github.com/tensorflow/community/pull/435
+
+Objective
+
+The aim of this RFC is to:
+
+1. Any exception/status code thrown by XLA to be uniquely named (independent of TensorFlow).
+2. XLA exceptions to be readily available to front-end frameworks so that they can be exposed to users.
+
+Motivation
+
+1. Though XLA was developed as a part of Tensorflow, it is also used by PyTorch, JAX, etc. There is a need for exceptions thrown by XLA to be named independent of Tensorflow. 
+2. Today, XLA defines XlaRuntimeError in the xla client. However, this exception is not readily available to front-ends. Refer to this (https://github.com/google/jax/pull/10676) effort in JAX to expose this Exception to users.
+
+User Benefit
+
+1. *Simplified contract with front-ends*: Add a single line of code to front-ends to import XLA exceptions. For eg: from import xla.exceptions as xla_exceptions.
+2. *Better exception names in XLA*: Derive readable exceptions like XlaInvalidArgumentError from XlaError to replace the existing ValueError. Additionally, this would make it easier to catch/trap Exceptions from XLA and attribute errors to XLA.
+
+Design Proposal
+
+This proposal would replace existing std:: and py:: exceptions thrown by the CPP layers with new custom exceptions which will all be extended from XlaError base class. All status objects originating from XLA will be replaced by XLA specific status codes. Equivalent python exceptions will be defined under a common module xla.exceptions which will contain translations for status objects (previously TensorFlow exceptions (https://github.com/tensorflow/tensorflow/blob/87462bfac761435a46641ff2f10ad0b6e5414a4b/tensorflow/python/framework/errors_impl.py)) as well as direct throws from XLA (previously std library exceptions).
+
+This proposal introduces an XlaError base class and all exceptions are extended from this class.
+
+class XlaError : public std::Exception {
+public:
+    char * what () {
+        return "Custom XLA Exception";
+    }
+}
+
+We will create different exception classes based on the need by extending this base class. For example:
```

Comment:
I assume initially "need" is defined as existing TensorFlow exceptions that are currently used? What is the intended use for this exception hierarchy? You could also achieve the goal you have defined by just having an `XlaError` as the only exception that is thrown at the interface boundary to XLA. That exception could carry a status code, message and stack trace to aid debugging.

I acknowledge that this is currently not very principled in which exception is thrown but if we make all thrown exceptions XLA specific anyway, we could unify the use now. Hence I'd like to better understand the use case of sub-exceptions.